client/landing/stepper/declarative-flow/internals/steps-repository/goals/select-goals.tsx

```diff
@@ -18,40 +19,54 @@ export const SelectGoals: React.FC< SelectGoalsProps > = ( {
 	const translate = useTranslate();
 	const goalOptions = useGoals();
 
-	const handleChange = ( selected: boolean, value: Onboard.SiteGoal ) => {
-		const newSelectedGoals = [ ...selectedGoals ];
-		const goalKey = value;
+	const handleChange = ( selected: boolean, goal: Onboard.SiteGoal ) => {
+		// Always remove potential duplicates
+		const newSelectedGoals = selectedGoals.filter( ( selectedGoal ) => selectedGoal !== goal );
 
+		// Add newly selected goal to the array
 		if ( selected ) {
-			newSelectedGoals.push( goalKey );
-		} else {
-			const goalIndex = newSelectedGoals.indexOf( goalKey );
-			newSelectedGoals.splice( goalIndex, 1 );
+			newSelectedGoals.push( goal );
 		}
-
 		onChange( newSelectedGoals );
 	};
 
+	const handleImportLinkClick = () => {
+		// Internally, pick Import as a selected goal.
+		// In goalsToIntent(), it will choose import as intent.
+		handleChange( true, Onboard.SiteGoal.Import );
+		onSubmit();
+	};
+
+	const handleContinueButtonClick = () => {
+		// Omit import from selection if it was added to the selection
```

Comment:
Instead of doing the clean up process here, it feels to me like it would be more correct to do so on component's initial mount, i.e. `useEffect( cleanupFn, []);`.

Reply:
Okay try that. :)